docs/development.md

```diff
@@ -134,6 +141,48 @@ in your boxes.yaml entry to configure this on box creation.
 #### Additional info
 - `foreman start` can be run with `rails` or `webpack` to start just that server: i.e. `bundle exec foreman start rails`. This can be used to start servers separately.
 
+### Customizing the Development Environment
+
+#### Custom local files
+
+You can have files automatically copied over to the working user's home directory (e.g. `~/vagrant`) when spinning up a `centos7-katello-devel` or `centos7-katello-devel-stable` box. To do so, add any files you want to be copied over to your development box to `user_devel_env_files/` found in forklift's root directory
+
+The directory `user_devel_env_files/` is ignored by git so the files won't be checked into version control. The files added to `user_devel_env_files/` will be copied over to `~/vagrant` on the development VM when it is created or provisioned.
+
+As an example, you can keep dotfiles such as `.bashrc` in this directory.
+
+#### Custom files from git repo
+
+A git repo's contents can be copied to the working user's home directory (e.g. `~/vagrant`) when spinning up a `centos7-katello-devel` or `centos7-katello-devel-stable` box. This can be done by specifying the `customize_home_git_repo` ansible variable. For example:
+
+```
+centos7-katello-devel-stable:
+  box_name: katello/katello-devel
+  hostname: centos7-katello-devel-stable.example.com
+  ansible:
+    playbook: 'playbooks/setup_user_devel_environment.yml'
+    variables:
+      customize_home_git_repo: https://github.com/johnpmitsch/config_settings
+```
+
+The contents of the specified repo will be copied to the home directory excluding the `.git` folder and `.gitignore` file.
```

Comment:
If I supply a bootstrap script, does this still perform the copy (I would assume it should not)? That should be made clear to the user.

Reply:
@ehelms the order of operations is:

- Git repo's contents is dumped in to the target user's home dir
- .git and .gitignore files are removed if present
- the specified bootstrap script script is run, or if not specified, a `bootstrap` script is run if found. If nothing is specified and not `bootstrap` script present, do nothing.

So if I understand the question correctly, the copy always happens regardless of whether the bootstrap script is present. Does that answer your question? I can clarify in the docs, but trying to understand the comment a little better.

Reply:
That does clarify and might cause errors. I suppose I can test. I sorta liked the idea of its either or but not both. My bootstrap script for example symlinks my dotfiles so when I git pull updates, they are automatically in place. I can understand if I am at a bit of an advanced use case point,

Reply:
> That does clarify and might cause errors. I suppose I can test. I sorta liked the idea of its either or but not both. My bootstrap script for example symlinks my dotfiles so when I git pull updates, they are automatically in place. I can understand if I am at a bit of an advanced use case point,

That is one thing that did come to mind, I actually do the same (or did, before I updated for this PR). However, you could `vagrant provision` to apply the latest updates in the git repo - especially with the stable box, this will only run the customization steps since its a pre-installed image. Its a different workflow, but I think for the better than trying to manage symlinked dotfiles on a devel box.

We get some feedback from devs once this is merged and see if there are concerns or asks in this area. Its definitely more of a "convention over configuration" approach, but I think trying to adapt to everyone's workflows could be a slippery slope and add add too much complexity.

Reply:
@ehelms  Also, I want to add, the current procedure isn't exclusive to dotfiles, thats just the intended usage so it makes for a good example. It is just dumping a git repo into the home directory and running a script if its present, so you can make that work for you however you want. 

It seems like you are thinking of replicating the cloning of the repo and symlinking the dotfiles strategy. So the big difference is the copying vs. the cloning. I think the copying is simpler (especially since you can re-provision to updated) so it would be nice to start out that way